src/main/java/uk/ac/ebi/ampt2d/metadata/exceptionhandling/ExceptionHandlers.java

```diff
@@ -29,4 +32,13 @@
     public ResponseEntity<ErrorMessage> handleIllegalArgumentException(IllegalArgumentException ex) {
         return new ResponseEntity(new ErrorMessage(HttpStatus.BAD_REQUEST, ex, ex.getMessage()), HttpStatus.BAD_REQUEST);
     }
+
+    @ExceptionHandler(value = MethodArgumentTypeMismatchException.class)
+    public ResponseEntity<ErrorMessage> handleMethodArgumentTypeMismatchException(MethodArgumentTypeMismatchException ex) {
+        if ( ex.getRequiredType() == LocalDate.class ) {
+            return handleIllegalArgumentException(new IllegalArgumentException("Please provide a date in the form yyyy-mm-dd"));
+        }
+
+        throw ex;
```

Comment:
Would not be catched again by the advice?

Reply:
No, it would not be caught by the same handler again. It will be caught by other handlers if there is any.